tests/test_connection.py

```diff
@@ -64,11 +64,9 @@ def test_Timeout_and_Deadline(self):
         yield self.named_conn.db.coll.insert({'x': 42}, safe=True, timeout=10)
         yield self.named_conn.db.coll.insert({'x': 42}, safe=True, deadline=time()+10)
 
-        d_insert = self.named_conn.db.coll.insert({'x': 42}, safe=True, deadline=time()-10)
-        yield self.assertFailure(d_insert, TimeExceeded)
+        self.assertRaises(TimeExceeded, self.named_conn.db.coll.insert, {'x': 42}, safe=True, deadline=time()-10)
 
-        d_insert = self.named_conn.db.coll.insert({'x': 42}, safe=True, timeout=-10)
-        yield self.assertFailure(d_insert, TimeExceeded)
+        self.assertRaises(TimeExceeded, self.named_conn.db.coll.insert, {'x': 42}, safe=True, timeout=-10)
```

Comment:
I'm not sure... Previously all exceptions from `@inlineCallbacks`-based methods were returned as failed Deferred. And calling code might be sure that it will always return Deferred. This PR changes this: for some errors it will still return failed Deferreds, but other errors might be raised immediately as direct exceptions. Hence all these changes in testing code.

This actually changes current API and potentially may break some existing code. (inlineCallbacks-based user code won't be affected). 

Previous behavior can be easily saved by returning `defer.fail(...)` instead of raising, but it looks ugly.

What do you think?